gammapy/datasets/map.py

```diff
@@ -1128,6 +1134,59 @@ def from_dict(cls, data, lazy=False, cache=True):
         dataset = cls.read(filename, name=data["name"], lazy=lazy, cache=cache)
         return dataset
 
+    @classmethod
+    def from_images(cls, images, energy_min, energy_max, name=None, **kwargs):
+        """Create a MapDataset from a set 2D images which do not have an
+        energy axis. An appropriate energy axis is added for all the images.
+
+        This can be useful for modelling and fitting data reduced outside gammapy
+
+        Parameters
+        ----------
+        images: dict
+        Dictionary of images. Allowed keywords are `counts`, `background`,
+        `exposure` and `psf`. The psf should be a psf kernel
+        energy_min : `~astropy.units.Quantity`
+            Minimum energy bound of the maps
+        energy_max :`~astropy.units.Quantity`
+            Maximum energy bound of the maps
+        name : str
+            Name of the returned dataset.
+        kwargs: kwargs passed to `MapDataset`
+
+        Returns
+        --------
+        dataset: `MapDataset`
+            Map dataset.
+        """
+
+        name = make_name(name)
+        kwargs["name"] = name
+
+        energy_axis = MapAxis.from_energy_edges([energy_min, energy_max], name="energy")
+        energy_axis_true = MapAxis.from_energy_edges(
```

Comment:
What if the user has a different binning for psf kernel and exposure?

Reply:
But we always require that the PSF kernel and exposure have the same energy axis, except if the PSF is in reco energy. Is that what you meant?